### PR TITLE
PUBDEV-6119  add tests for predictions with stratification

### PR DESCRIPTION
**Pull request by satai**
`stratify_by` seems to produce a different result than in R.

### CODE REVIEW

File: h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R [hunk 1]
```diff
@@ -2,99 +2,153 @@ setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
...
-        lp[!is.na(lp)]
-    }
-
-    # Case 1: categoricals, numeric + interactions
-    tstdata.1 <- cancer
-    tstdata.1$sex <- as.factor(tstdata.1$sex)
+   
+
+    ## Case 1: categoricals, numeric + interactions
+    tstdata.1 <- cancer.with.sex()
```
Comment (honzasterba) [inline h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R:32]: I dont think you need tstdata.N, I think you should be fine with one frame being modified for each test if necessary

File: h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R [hunk 2]
```diff
@@ -2,99 +2,153 @@ setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
 source("../../../scripts/h2o-r-test-setup.R")
 
 
+pred.r <- function(fit.8, tstdata.8) {
```
Comment (honzasterba) [inline h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R:5]: param names indicate this has something to do with cast #8, but I dont think thats the case, samoe for compare.results

Comment (satai) [reply h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R:5]: You are definitely right, fixing this.

File: h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R (hunk 1 above)
Comment (satai) [reply h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R:32]: Technicaly, there is no need for it.

I would like to keep it this way anyway. Now is each test case structured in given-when-then feshion without sideeffect on data (and data is shared only when needed for comparisons - such as in the case that checks that equal weights don't afect the result).

**Pull request merged**